core/src/main/scala/com/banno/cosmos4s/types/QueryOptions.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 Jack Henry & Associates, Inc.®
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.banno.cosmos4s.types
+
+import com.azure.cosmos.models.CosmosQueryRequestOptions
+import cats.implicits._
+
+final class QueryOptions private (
+    val maxDegreeOfParallelism: Option[Int],
+    val maxBufferedItemCount: Option[Int]
+)
```

Comment:
To be a contraband class this requires a private copy method which copies all the current values over.

Then have with methods that delegate to the private method.

Methods have a much better binary compatibility profile.

Reply:
https://github.com/Banno/cosmos4s/pull/4/commits/44f91fd94a1ec77ba658153803c150fc1205b932